spring-integration-core/src/test/java/org/springframework/integration/store/SimpleMessageStoreTests.java

```diff
@@ -148,4 +151,19 @@ public void execute(MessageGroupStore messageGroupStore, MessageGroup group) {
 
 	}
 
+	@Test
+	public void testAddAndRemoveMessagesFromMessageGroup() throws Exception {
+		SimpleMessageStore messageStore = new SimpleMessageStore();
+		String groupId = "X";
+		List<Message<?>> messages = new ArrayList<Message<?>>();
+		for (int i = 0; i < 25; i++) {
+			Message<String> message = MessageBuilder.withPayload("foo").setCorrelationId(groupId).build();
+			messageStore.addMessageToGroup(groupId, message);
+			messages.add(message);
+		}
+		messageStore.removeMessagesFromGroup(groupId, messages);
+		MessageGroup group = messageStore.getMessageGroup(groupId);
+		assertEquals(0, group.size());
```

Comment:
This test doesn't test `Add` premise :smile: .
Joke, of course, but I haven't been sure at a glance that `removeMessagesFromGroup` had done anything with store at all...